glmtools/glm4d.py

```diff
@@ -53,5 +64,34 @@ def t_test_3d(c, X, B, sigma_2, df):
     p : array shape (V,)
         two-tailed probability value for each t statistic.
     """
-    # Your code code here
-    return
+    #reshape for t-test loop
+    B_2D = B.reshape(-1, B.shape[-1]).T
+    sigma_2_1D = sigma_2.reshape(np.prod(sigma_2.shape))
+    #loop through every voxel
+    #for i in range(np.prod(sigma_2.shape)):
+    #    t, p = t_test(c, X[:,i], B_2D[:,i], sigma_2[i], df)
+    #    t_array[i] = t
+    #    p_vec[i] = p
+    t, p = t_test(c, X, B_2D, sigma_2_1D, df)
+    #reshape
+    t_3D = t.reshape(Y.shape[0],Y.shape[1],Y.shape[2])
+    p_3D = p.reshape(Y.shape[0],Y.shape[1],Y.shape[2])
+
+    return t_3D, p_3D
+
+def test_glm4d_t_test3d():
+    #generate fake data to test functions
+    n = 20
+    x = np.random.normal(10, 2, size=n**3)
+    X = np.ones((n**3, n))
+    X[:, 1] = x
+    X = X.T
+    Y = np.random.normal(20, 1, size=(n,n,n,n))
+
+    B, sigma_2, df = glm_4d(Y,X)
+    c = np.ones(n**3)
```

Comment:
Your `c` here is puzzling.  Don't forget the `c` is to weight parameters to test.  For a t test, it needs to be a 1D vector, or a 2D column vector, with the same number of elements as there are columns in the design, and therefore the same number of elements are there are rows in the beta matrix.